src/core/processing/processingalgorithmsmodel.cpp

```diff
@@ -0,0 +1,261 @@
+/***************************************************************************
+  processingalgorithmsmodel.h - ProcessingAlgorithmsModel
+
+ ---------------------
+ begin                : 18.06.2024
+ copyright            : (C) 2024 by Mathieu Pellerin
+ email                : mathieu at opengis dot ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#include "processingalgorithmsmodel.h"
+
+#include <qgsapplication.h>
+#include <qgsnativealgorithms.h>
+#include <qgsprocessingalgorithm.h>
+#include <qgsprocessingprovider.h>
+#include <qgsprocessingregistry.h>
+#include <qgsvectorlayer.h>
+
+#include <mutex>
```

Comment:
just wondering: what for?

Reply:
Needed for the std::call_once lines in the base model class, i.e.:

```
  static std::once_flag initialized;
  std::call_once( initialized, [] { QgsApplication::processingRegistry()->addProvider( new QgsNativeAlgorithms( QgsApplication::processingRegistry() ) ); } );
```